src/components/tx/SignOrExecuteForm/ExecuteForm.tsx

```diff
@@ -50,8 +51,6 @@ const ExecuteForm = ({
   const { needsRiskConfirmation, isRiskConfirmed, setIsRiskIgnored } = useContext(TxSecurityContext)
 
   // Check that the transaction is executable
-  const isCreation = safeTx?.signatures.size === 0
-  const isNewExecutableTx = useImmediatelyExecutable() && isCreation
```

Comment:
I think it makes sense to not check for immediate execution here because if its a creation and its possible to execute it has to be an immediate execution so only checking the `isCreation` flag should be enough.

It would be nice to extract the error displaying logic into its own "dumb" component so we could test these cases though. I suggest to add a TODO.

Reply:
Feel free to create a tech debt ticket, I'd like to avoid polluting the code with TODO comments.